server-common/src/main/java/org/apache/gravitino/server/authorization/GravitinoAuthorizer.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.server.authorization;
+
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.authorization.Privilege;
+
+/** Used for metadata authorization. */
+public interface GravitinoAuthorizer {
+
+  /**
+   * After instantiating the GravitinoAuthorizer, execute the initialize method to perform a series
+   * of initialization operations, such as loading privilege policies and so on.
+   */
+  void initialize();
+
+  /**
+   * Perform authorization and return the authorization result.
+   *
+   * @param userId the user id in use_meta table
+   * @param metadataType for example, CATALOG, SCHEMA,TABLE, etc.
+   * @param metadataId the metadata id.
+   * @param privilege for example, CREATE_CATALOG, CREATE_TABLE, etc.
+   * @return authorization result.
+   */
+  boolean authorize(
+      long userId, MetadataObject.Type metadataType, long metadataId, Privilege.Name privilege);
```

Comment:
`metadataType` or  `metadataObject`?   seems we should authorize for the specific metadataObject

Reply:
> metadataObject

The "metadataObject" should be the corresponding "metadataId". In the authentication policy, "id" is used to refer to the "metadataObject".

Reply:
Where can we get the user id?

Reply:
> Where can we get the user id?

Get the username through PrincipalUtils and then query the userid from the database.

Reply:
Is there any reason for using `userId` not `userName` as the parameter?

Reply:
Because Gravitno associates users and roles in the `user_role_rel` table by using the `USER id` and `role id`.